### PR TITLE
Core level load mechanic

### LINKED ISSUE #99: Add game core architecture
The LevelTest scene was fine until now, but it's time to implement a serious code structure, and add the ability to load and unload levels.

We need a node to load and unload levels, code to place the player at the start of the level

### PR DESCRIPTION
**Pull request by razcore-art**
I've fiddled with the existing nodes and used Level1 & Level2 as
test beds. Level2 was actually named Level3, but I renamed some
some nodes to keep consistency.

The idea is very simple, there's an Area2D (called Exit) that
behaves like a portal, which has an export variable that points
to the next level (on the disk) to be loaded.

When the Player is detected by the Area2D `LevelLoader.trigger` gets
called.

LevelLoader is now an autoloaded script which keeps track of the game
node, the player node and the current level.

When the game first starts the game node also uses LevelLoader to load
the start level, instead of having it present in the scene tree by
default.

**note**: I had to mess with the nodes like I said so let me know if there's
any conflicts I can help with.

closes #99

### CODE REVIEW

**Comment by razcore-art** [comment]
@NathanLovato I also commited `Hook.gd` by mistake, don't include that in the commit, or... is there a way to uncommit it? :). That's part of another "fix" I was working on since I noticed some small issues

File: hook!-full-game/src/Levels/Exit.gd [hunk 1]
```diff
@@ -0,0 +1,25 @@
+tool
+extends Area2D
+
+
+export(String, FILE) var next_level_path: String = ""
```
Comment (NathanLovato) [inline hook!-full-game/src/Levels/Exit.gd]: Have you tried using a PackedScene directly? Using PackedScene we could drag and drop the levels into the inspector's slot. What do you think is best?

Not sure which is best. Also `next_level_path: String = ""` -> `next_level_path: = ""`, no need for the type hint

File: hook!-full-game/src/Main/Game.gd [hunk 2]
```diff
@@ -1,5 +1,11 @@
 extends Node
 
+const StartLevel: PackedScene = preload("res://src/Levels/Level1.tscn")
```
Comment (NathanLovato) [inline hook!-full-game/src/Main/Game.gd]: Can you make it an exported var? So we can swap levels for testing.

File: hook!-full-game/src/Player/Player.gd [hunk 3]
```diff
@@ -27,9 +27,15 @@ var info_dict: = {} setget set_info_dict
 
 
 func _ready() -> void:
+	Events.connect("level_loaded", self, "_on_Events_level_loaded")
 	stats.connect("health_depleted", state_machine, "transition_to", ['Die'])
 
 
+func _on_Events_level_loaded(level: Node2D) -> void:
+	var spawn_checkpoint = level.get_node("Checkpoints").get_child(0)
```
Comment (NathanLovato) [inline hook!-full-game/src/Player/Player.gd]: The player shouldn't access the checkpoint like that imo. Guilherme has worked on a checkpoints node that keeps track of visited checkpoints, we should go through something like that. Although maybe it's a job for another task or PR

File: hook!-full-game/src/Levels/Exit.gd (hunk 1 above)
Comment (razcore-art) [reply hook!-full-game/src/Levels/Exit.gd]: Doesn't work with PackedScene directly if we want to do cyclic paths: `Level1`'s `Exit` points to `Level2` & `Level2`'s `Exit` points to `Level1` wouldn't work, that's why I ended up  with `export(String, FILE)`

File: hook!-full-game/src/Player/Player.gd (hunk 3 above)
Comment (razcore-art) [reply hook!-full-game/src/Player/Player.gd]: Yeah I agree, I think this operation can be moved in `LevelLoader`, it should know how to do player setup/positioning as well

File: hook!-full-game/src/Levels/Exit.gd (hunk 1 above)
Comment (NathanLovato) [reply hook!-full-game/src/Levels/Exit.gd]: Got it!

**Pull request merged**